### PR TITLE
[5.8] Use Name of Factory as Model

### PR DESCRIPTION
**Pull request by simoebenhida**
Use the name of factory as the model if `--model` is not specified
```
php artisan make:factory UserFactory
```

the Model will be `User` here

### CODE REVIEW

File: src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ protected function buildClass($name)
     {
         $namespaceModel = $this->option('model')
                         ? $this->qualifyClass($this->option('model'))
-                        : trim($this->rootNamespace(), '\\').'\\Model';
+                        : trim($this->rootNamespace(), '\\').'\\'. ucfirst(str_replace('Factory', '', $name));
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: Replacing every occurrence of Factory is not what we want surely. What about a model actually being called `Factory`? Surely we should instead just delete the last 7 characters if they are `Factory`?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php (hunk 1 above)
Comment (simoebenhida) [reply src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: Yeah good point !! what if there is no `Factory` and we deleted the model name i think we should check if the string has a `Factory` string ?

**Comment by simoebenhida** [comment]
This is will works when you only enter `Factory` or when you write a name with no `Factory` 
```! strpos($name, 'Factory') ? 'Model' : ucfirst(str_replace('Factory', '', $name));```

**Comment by crnkovic** [comment]
IMO, deleting the last 7 characters is “more correct”, and that was your intention also :)

Good PR, hopefully will make it into the core.

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient **explanation**, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Pull request closed without merging**